linting/dockerLinting.ts

```diff
@@ -0,0 +1,91 @@
+import vscode = require('vscode');
+import { diagnosticCollection } from '../dockerExtension';
+import * as fs from 'fs';
+import * as path from 'path';
+
+var DockerFileValidator = require('dockerfile_lint');
+let configOptions: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration('docker');
+
+function createErrDiagnostic(e, doc: vscode.TextDocument): vscode.Diagnostic {
+    var r: vscode.Range;
+    var lineNum = -1;
+    var msg = e.message;
+
+    if (e.line) {
+        lineNum = e.line;
+    }
+
+    if (lineNum === -1) {
+        r = new vscode.Range(0, 0, 0, 0);
+    } else {
+        r = doc.lineAt(lineNum - 1).range
+    }
+
+    if (e.lineContent) {
+        msg = msg + ': ' + e.lineContent;
+    }
+
+    return new vscode.Diagnostic(r, msg, vscode.DiagnosticSeverity.Error);
+
+}
+
+export function scheduleValidate(document: vscode.TextDocument) {
+    let urisToValidate: { [uri: string]: boolean; } = {};
+    let timeoutToken: NodeJS.Timer = null;
+
+    if (document.languageId !== 'dockerfile') {
+        return;
+    }
+
+    urisToValidate[document.uri.toString()] = true;
+
+    if (timeoutToken !== null) {
+        clearTimeout(timeoutToken);
+    }
+
+    timeoutToken = setTimeout(() => {
+        timeoutToken = null;
+        vscode.workspace.textDocuments.forEach((document) => {
+            if (urisToValidate[document.uri.toString()]) {
+                doValidate(document);
+            }
+        });
+
+        urisToValidate = {};
+    }, 200);
+
+    return;
+}
+
+function doValidate(document: vscode.TextDocument) {
+
+    let diagnostics: vscode.Diagnostic[] = [];
+
+    let linterRuleFile = configOptions.get('linterRuleFile', 'basic_rules.yaml');
+
```

Comment:
It looks like this lib will actually default to using this exact rules file if you specific a null/empty rules file path, so we could choose not to ship any of these sample rules file, and allow either turning on the default rules or specifying a custom one.